### PR TITLE
REF: collect+parametrize reorder_levels tests

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/generic/methods/test_reorder_levels.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        tm.assert_equal(result, expected)
+
+    def test_reorder_levels_swaplevel_equivalence(self):
+
+        ymd = tm.makeTimeDataFrame(100)
+        mi = MultiIndex.from_arrays(
+            [ymd.index.year, ymd.index.month, ymd.index.day],
+            names=["year", "month", "day"],
+        )
+        ymd.index = mi
```
Comment (simonjayhawkins) [inline pandas/tests/generic/methods/test_reorder_levels.py]: is this the same as multiindex_year_month_day_dataframe_random_data fixture?

File: pandas/tests/generic/methods/test_reorder_levels.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            levels=[["bar"], ["one", "two", "three"], [0, 1]],
+            codes=[[0, 0, 0, 0, 0, 0], [0, 1, 2, 0, 1, 2], [0, 1, 0, 1, 0, 1]],
+            names=["L0", "L1", "L2"],
+        )
+        df = DataFrame({"A": np.arange(6), "B": np.arange(6)}, index=index)
+        obj = df if klass is DataFrame else df["A"]
+
+        # no change, position
+        result = obj.reorder_levels([0, 1, 2])
+        tm.assert_equal(obj, result)
```
Comment (simonjayhawkins) [inline pandas/tests/generic/methods/test_reorder_levels.py]: if this is going to happen a lot with these changes, should maybe consider a solution that can maintain the correct assertion helper.

either in the parameterisation as "klass, helper" or create a fixture something like

```
@fixture()
def assert_klass_equal(klass):
    if isinstance(klass, DataFrame):
      return tm.assert_frame_equal
    elif isinstance(klass, Series):
      return tm.assert_series_equal
   else:
      assert False 
```
(code not tested!)

Comment (jbrockmendel) [reply pandas/tests/generic/methods/test_reorder_levels.py]: might make sense to add as a kwarg to tm.assert_equal

File: pandas/tests/generic/methods/test_reorder_levels.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/generic/methods/test_reorder_levels.py]: yes, will update

File: pandas/tests/generic/methods/test_reorder_levels.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/tests/generic/methods/test_reorder_levels.py]: yeah, that sounds good as well. could be follow-up

**Review by simonjayhawkins — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**